internal/plan/plan.go

```diff
@@ -258,6 +259,88 @@ const (
 	ActionIgnore   ServiceAction = "ignore"
 )
 
+type LogTargets struct {
+	Targets []string
+	Keyword string
+}
+
+const (
+	// Valid keywords for LogTargets
+	DefaultLogTargets = "default"
+	AllLogTargets     = "all"
+	NoLogTargets      = "none"
+)
+
+func (t *LogTargets) UnmarshalYAML(value *yaml.Node) error {
+	if value.Kind == yaml.ScalarNode {
+		switch value.Value {
+		case DefaultLogTargets, AllLogTargets, NoLogTargets:
+			t.Keyword = value.Value
+			return nil
+		default:
+			return fmt.Errorf("invalid value %q for LogTargets", value.Value)
+		}
+	}
+
+	// unmarshal to []string
+	return value.Decode(&t.Targets)
+}
+
+// Copy returns a deep copy of this LogTargets struct.
+func (t *LogTargets) Copy() *LogTargets {
+	return &LogTargets{
+		Targets: append([]string(nil), t.Targets...),
+		Keyword: t.Keyword,
+	}
+}
+
+// Merge merges the fields from other into t.
+func (t *LogTargets) Merge(other *LogTargets) {
+	// anything <- keyword yields keyword
+	if other.Keyword != "" {
+		t.Keyword = other.Keyword
+		t.Targets = nil
+	}
+
+	// keyword <- [list] yields [list]
+	if t.Keyword != "" && len(other.Targets) > 0 {
```

Comment:
I think ```UnmarshalYAML``` can only ever result in either a keyword, or a log target list, so the logic seems to work. However, it feels to me that these cases should be mutually exclusive, so would it not be clearer to explicitly enforce that using ```if {} else if ... ``` ? Maybe I am missing a case.

Reply:
you're right. I'll try to make this more clear